From a03cede653fa5f43e0519ca7b74ee251967e3019 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Thu, 10 Oct 2013 19:32:48 -0700
Subject: [PATCH] Revert "Torch: only write to sysfs if not already on"

This reverts commit 090cd7927a190091fb7d4e7c334b617252fa8365.

Allows Torch to work when the screen is turned off.

To quote Dave Daynard, "This patch *only* applies to devices without a dedicated torch mode and only a single brightness. There is an entire other section of code dedicated to devices with multiple torch modes which this patch does not touch."
This means that we could use the camera interface to access the flash LEDs in order to fix Torch when the screen is turned off, but using the camera interface uses more power. Our kernel's LED driver has a dedicated sysfs interface for enabling the flash LEDs, so just revert this commit.

http://review.cyanogenmod.org/#/c/49628/
---
 src/net/cactii/flash2/FlashDevice.java | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/net/cactii/flash2/FlashDevice.java b/src/net/cactii/flash2/FlashDevice.java
index ecbda15..ff753b8 100644
--- a/src/net/cactii/flash2/FlashDevice.java
+++ b/src/net/cactii/flash2/FlashDevice.java
@@ -216,11 +216,8 @@ public synchronized void setFlashMode(int mode) {
                     if (mFlashDeviceWriter == null) {
                         mFlashDeviceWriter = new FileWriter(mFlashDevice);
                     }
-                    // Write to sysfs only if not already on
-                    if (mode != mFlashMode) {
-                        mFlashDeviceWriter.write(String.valueOf(value));
-                        mFlashDeviceWriter.flush();
-                    }
+                    mFlashDeviceWriter.write(String.valueOf(value));
+                    mFlashDeviceWriter.flush();
                     if (mode == OFF) {
                         mFlashDeviceWriter.close();
                         mFlashDeviceWriter = null;
-- 
1.8.4

